ansible/roles/host-ocp4-hcp-cnv-install/tasks/main.yaml

```diff
@@ -0,0 +1,304 @@
+      until:
+        - r_hosted_cluster.resources[0].status.kubeconfig is defined
+        - r_hosted_cluster.resources[0].status.kubeconfig.name is defined
+        - r_hosted_cluster.resources[0].status.kubeconfig.name | length > 0
+
+    - name: Get the OpenShift admin kubeconfig secret
+      kubernetes.core.k8s_info:
+        api_version: v1
+        kind: Secret
+        name: hcp-{{ guid }}-admin-kubeconfig
+        namespace: "{{ hcp_ocp_namespace }}"
+      register: kubeadmin_secret_result
+
+    - name: Get the OpenShift secret
+      when: hcp_authentication | default('') != 'htpasswd'
+      kubernetes.core.k8s_info:
+        api_version: v1
+        kind: Secret
+        name: hcp-{{ guid }}-kubeadmin-password
+        namespace: "{{ hcp_ocp_namespace }}"
+      register: password_secret_result
+
+    - name: Decode the kubeconfig secret
+      set_fact:
+        kubeconfig_decoded: "{{ kubeadmin_secret_result.resources[0].data.kubeconfig | b64decode }}"
+
+    - name: Decode the kubeadmin password secret
+      when: hcp_authentication | default('') != 'htpasswd'
+      set_fact:
+        kubeadmin_password: "{{ password_secret_result.resources[0].data.password | b64decode }}"
+
+    - name: Generate /tmp/kubeconfig
+      copy:
+        dest: /tmp/kubeconfig
+        content: "{{ kubeconfig_decoded }}"
+
+- name: Get information about the router-nodeport-default service
+  kubernetes.core.k8s_info:
+    kubeconfig: /tmp/kubeconfig
+    api_version: v1
+    kind: Service
+    namespace: openshift-ingress
+    name: router-nodeport-default
+  register: service_info
+  retries: 10
+  delay: 30
+  until: service_info.resources[0].spec.ports is defined and service_info.resources[0].spec.ports | length > 0
+
+- name: Extract nodeport for http
+  set_fact:
+    nodeport_http: "{{ service_info.resources[0].spec.ports | selectattr('name', 'equalto', 'http') | map(attribute='nodePort') | first }}"
+
+- name: Extract nodeport for https
+  set_fact:
+    nodeport_https: "{{ service_info.resources[0].spec.ports | selectattr('name', 'equalto', 'https') | map(attribute='nodePort') | first }}"
+
+- name: Post-Install tasks
+  module_defaults:
+    group/k8s:
+      host: "{{ sandbox_openshift_api_url }}"
+      api_key: "{{ k8s_auth_results.k8s_auth.api_key|default(sandbox_openshift_api_key) }}"
+      validate_certs: false
+
+  block:
+    - name: Add the service (type LoadBalancer) for the apps
+      kubernetes.core.k8s:
+        definition: "{{ lookup('ansible.builtin.template', 'workers_svc.yaml') }}"
+        wait: true
+        wait_timeout: 300
+      vars:
+        namespace: "{{ hcp_ocp_namespace }}"
+
+    - name: Wait for the LoadBalancer value - workers
+      register: svc_apps
+      kubernetes.core.k8s_info:
+        api_version: v1
+        kind: Service
+        name: "svc-{{ guid }}-apps"
+        namespace: "{{ hcp_ocp_namespace }}"
+      until: svc_apps.resources[0].status.loadBalancer.ingress[0].ip | default('') != ''
```

Comment:
I think that's OK. I usually write | length > 0. But the end result is probably the same.